include/lf.h

```diff
@@ -59,19 +59,19 @@ typedef struct {
   lf_pinbox_free_func *free_func;
   void *free_func_arg;
   uint free_ptr_offset;
-  uint32 volatile pinstack_top_ver;         /* this is a versioned pointer */
-  uint32 volatile pins_in_array;            /* number of elements in array */
+  uint64 volatile pinstack_top_ver;         /* this is a versioned pointer */
+  uint64 volatile pins_in_array;            /* number of elements in array */
 } LF_PINBOX;
 
 typedef struct st_lf_pins {
   void * volatile pin[LF_PINBOX_PINS];
   LF_PINBOX *pinbox;
   void  *purgatory;
-  uint32 purgatory_count;
-  uint32 volatile link;
+  uint64 purgatory_count;
+  uint64 volatile link;
 /* we want sizeof(LF_PINS) to be 64 to avoid false sharing */
-#if SIZEOF_INT*2+SIZEOF_CHARP*(LF_PINBOX_PINS+2) != 64
-  char pad[64-sizeof(uint32)*2-sizeof(void*)*(LF_PINBOX_PINS+2)];
+#if SIZEOF_INT*4+SIZEOF_CHARP*(LF_PINBOX_PINS+2) != 64
```

Comment:
Is the SIZEOF_INT dependency correct here, and why it's needed in the first place when we use exact-width type? I.e. can SIZEOF_INT*4 can be replaced with 16?

Reply:
It can be, I'll change it.

Reply:
I updated this part.

I also didn't add any mtr test yet - based on Friday's meeting.